sdk/attestation/azure-security-attestation/src/attestation_administration_client.cpp

```diff
@@ -259,20 +261,230 @@ AttestationAdministrationClient::ResetAttestationPolicy(
+  return Response<AttestationToken<Models::PolicyResult>>(returnedToken, std::move(response));
+}
+
+Azure::Response<Models::AttestationToken<PolicyCertificateListResult>>
+AttestationAdministrationClient::GetPolicyManagementCertificates(
+    GetPolicyManagementCertificatesOptions const& options,
+    Azure::Core::Context const& context) const
+{
+  auto request = AttestationCommonRequest::CreateRequest(
+      m_endpoint, m_apiVersion, HttpMethod::Get, {"certificates"}, nullptr);
+
+  // Send the request to the service.
+  auto response = AttestationCommonRequest::SendRequest(*m_pipeline, request, context);
+
+  // Deserialize the Service response token and return the JSON web token returned by the
+  // service.
+  std::string responseToken = AttestationServiceTokenResponseSerializer::Deserialize(response);
+
+  // Parse the JWT returned by the attestation service.
+  auto resultToken = AttestationTokenInternal<
+      Models::_detail::GetPolicyCertificatesResult,
+      PolicyCertificateGetResultSerializer>(responseToken);
+
+  // Validate the token returned by the service. Use the cached attestation signers in the
+  // validation.
+  std::vector<AttestationSigner> const& signers = GetAttestationSigners(context);
+  resultToken.ValidateToken(
+      options.TokenValidationOptions ? *options.TokenValidationOptions
+                                     : this->m_tokenValidationOptions,
+      signers);
+
+  Models::_detail::JsonWebKeySet jwks(
+      *static_cast<AttestationToken<Models::_detail::GetPolicyCertificatesResult>>(resultToken)
+           .Body.PolicyCertificates);
+  Models::PolicyCertificateListResult returnedResult;
+  for (const auto& certificate : jwks.Keys)
   {
-    returnedResult.PolicySigner = AttestationSignerInternal(*internalResult.PolicySigner);
+    returnedResult.Certificates.push_back(AttestationSignerInternal(certificate));
   }
-  if (internalResult.PolicyTokenHash)
+
+  // Construct a token whose body is the get policy certificates result, but whose token is the
+  // response from the service.
+  auto returnedToken = AttestationTokenInternal<Models::PolicyCertificateListResult>(
+      responseToken, returnedResult);
+  return Response<AttestationToken<Models::PolicyCertificateListResult>>(
+      returnedToken, std::move(response));
+}
+
+Azure::Response<Models::AttestationToken<Models::PolicyCertificateModificationResult>>
+AttestationAdministrationClient::AddPolicyManagementCertificate(
+    std::string const& pemEncodedX509CertificateToAdd,
+    AttestationSigningKey const& existingSigningKey,
+    AddPolicyManagementCertificatesOptions const& options,
+    Azure::Core::Context const& context) const
+{
+  // Calculate a signed (or unsigned) attestation policy token to send to the service.
+  // Embed the encoded policy in the StoredAttestationPolicy.
+  auto x5cToAdd(Cryptography::ImportX509Certificate(pemEncodedX509CertificateToAdd));
```

Comment:
const?

Reply:
Using `const` for local variables which we expect not to be updated anywhere after they are created is for buying an insurance policy for the future :). In other words, for preventing bugs, like someone adding a change in the future where the variable is mutated.  When not `const`, there's not any hint for future developers/maintainers to let them know that the intention for that variable is to be a read-only.

You can add a comment as well, but it is not as effective as blocking with `const`.

So, it's not for performance or compiler-oriented.  This is for self-documented code.  The author makes very clear about the intention of a variable as `not to be updated within the function`

Reply:
That's fair. My biggest concern here is that it leads to const poisoning, which is ... unfortunate.

For instance, correctly fixing this issue is going to require changing the function signature of the `HttpPipeline::Send`, which is NOT going to be pretty (`HttpPipeline::Send` requires a mutable reference to the `Request` object, but there's no reason for the HttpPipeline to ever modify the Request object.